### DISCUSSION TITLE
How to popup a window to show the Phantom swap page on desktop browser?

### DISCUSSION THREAD

**Discussion by stellan-wu** [phantom/docs#309] [Help & Support] [answered]
### Summary

Hello, Phantom experts
Hope you are all doing well.

I have a web page that is opened in Chrome browser, I already installed the Phantom extension, and there is a link or button on the page, when clicking this link, I want the Phantom wallet extension can show a popup window to show the swap page, which tokens should be prefilled in the "Pay" and "Receive" fields.
This behavior looks like the deeplink feature in the Phantom mobile app. Does the Phantom Chrome extension support this behavior?

Thank you.

### Steps to Reproduce

<img width="730" alt="image" src="https://github.com/user-attachments/assets/ea3ee58f-fd86-4065-acb8-25753e49dff4">

### Phantom Version

24.22.1

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment, marked as answer]
Hey @stellan-wu,

Not currently. We don't have an eta on our embedded wallet but I believe we will have more robust apis for actions like this within the Phantom embedded wallet, eventually. But for now you should build your own swap components.

**Reply by felipeandradevalenzuela** [reply]
is there a way to do it this year, or i should try making my website an pwa?
